packages/jazz-tools/src/coValues/group.ts

```diff
@@ -152,43 +147,44 @@ export class Group extends CoValueBase implements CoValue {
   }
 
   get members(): Array<{
-    id: ID<Account> | "everyone";
+    id: ID<RegisteredAccount>;
     role: Role;
-    ref: Ref<Account> | undefined;
-    account: Account | null | undefined;
+    ref: Ref<RegisteredAccount>;
+    account: RegisteredAccount;
   }> {
     const members = [];
 
+    const BaseAccountSchema =
+      (activeAccountContext.maybeGet()?.constructor as typeof Account) ||
+      RegisteredSchemas["Account"];
+    const refEncodedAccountSchema = {
+      ref: () => BaseAccountSchema,
+      optional: false,
+    } satisfies RefEncoded<RegisteredAccount>;
+
     for (const accountID of this._raw.getAllMemberKeysSet()) {
       if (!isAccountID(accountID)) continue;
 
       const role = this._raw.roleOf(accountID);
-      const ref = new Ref<Account>(
-        accountID as unknown as ID<Account>,
+      const ref = new Ref<RegisteredAccount>(
+        accountID as unknown as ID<RegisteredAccount>,
         this._loadedAs,
-        this._schema[MembersSym],
+        refEncodedAccountSchema,
       );
-      const accessRef = () => ref?.accessFrom(this, "members." + accountID);
-
-      members.push({
-        id: accountID as unknown as ID<Account>,
-        role,
-        ref,
-        get account() {
-          return accessRef();
-        },
-      });
-    }
+      const accessRef = () => ref.accessFrom(this, "members." + accountID);
 
-    const everyoneRole = this._raw.roleOf("everyone");
```

Comment:
Removed everyone from the returned values to return a list of values of the same type making the usage of this API way easier.

For the cases where the dev wants to check for everyone access esplicitely they can do:

```ts
if (group.getRoleOf("everyone") {
  // Do stuff
}
```

It doesn't come to my mind a common use case for that so I think the extra code can be justified.

Reply:
Hmm I disagree about this one. Group.members would often be used to render the roles on a Group or something owned by it. Even though "everyone" makes the return type more awkward, it forces you to deal with and expose in the UI that case

Reply:
> Even though "everyone" makes the return type more awkward, it forces you to deal with and expose in the UI that case

That's not true, it only makes the UI code less readable without forcing you to do that (see https://github.com/garden-co/jazz/blob/main/examples/organization/src/components/OrganizationMembers.tsx#L15)

Having something this weird in the APIs that's hard to understand only leads devs to circumventing the type.

Reply:
OK - but this means we should instead make "everyone" behave more like an Account in the long run so we can return it as a member.

In the meantime the docs should shout out to check for "everyone"s role in Groups when displaying roles to the user